app/controllers/api/base_controller/renderer.rb

```diff
@@ -350,15 +352,19 @@ def attr_accessible?(object, attr)
       def attr_virtual?(object, attr)
         return false if ID_ATTRS.include?(attr)
         primary = attr_split(attr).first
-        (object.class.respond_to?(:reflect_on_association) && object.class.reflect_on_association(primary)) ||
-          (object.class.respond_to?(:virtual_attribute?) && object.class.virtual_attribute?(primary)) ||
-          (object.class.respond_to?(:virtual_reflection?) && object.class.virtual_reflection?(primary))
+        klass   = object
+        klass   = object.class if object.kind_of?(ActiveRecord::Base)
+        (klass.respond_to?(:reflect_on_association) && klass.reflect_on_association(primary)) ||
+          (klass.respond_to?(:virtual_attribute?) && klass.virtual_attribute?(primary)) ||
+          (klass.respond_to?(:virtual_reflection?) && klass.virtual_reflection?(primary))
       end
 
       def attr_physical?(object, attr)
         return true if ID_ATTRS.include?(attr)
-        (object.class.respond_to?(:has_attribute?) && object.class.has_attribute?(attr)) &&
-          !(object.class.respond_to?(:virtual_attribute?) && object.class.virtual_attribute?(attr))
+        klass = object
+        klass = object.class if object.kind_of?(ActiveRecord::Base)
+        (klass.respond_to?(:has_attribute?) && klass.has_attribute?(attr)) &&
+          !(klass.respond_to?(:virtual_attribute?) && klass.virtual_attribute?(attr))
```

Comment:
Are these cosmetic changes or is this changing behavior for performance reasons?  I'm ok with cosmetic changes but I don't see how this makes a meaningful performance gain, so I want to make sure I am understanding...

Reply:
(this block the similar change above it)

Reply:
these are not cosmetic - these seem to be working on `klass` instead of `object.class`

Reply:
> these are not cosmetic

These are cosmetic from a "`klass` and `object.class`" are the same, but the rational as to why I changed it because normally these methods took an instance (as it was traversing through the `jbuilder` methods... aka "N+1ing"), but this allows it to just pass the `klass` itself.

Reply:
> but the rational as to why I changed it because normally these methods took an instance (as it was traversing through the `jbuilder` methods... aka "N+1ing"), but this allows it to just pass the `klass` itself.

Thanks... that was the detail I was trying to understand.  👍